lib/logstash/plugin_mixins/opensearch/api_configs.rb

```diff
@@ -34,6 +34,20 @@ module APIConfigs
         # this defaults to a concatenation of the path parameter and "_bulk"
         :bulk_path => { :validate => :string },
 
+        # Maximum number of bytes in bulk requests
+        # The criteria for deciding the default value of target_bulk_bytes is:
+        # 1. We need a number that's less than 10MiB because OpenSearch is commonly
+        #    configured (particular in AWS Opensearch Service) to not accept
+        #    bulks larger than that.
+        # 2. It must be large enough to amortize the connection constant
+        #    across multiple requests.
+        # 3. It must be small enough that even if multiple threads hit this size
+        #    we won't use a lot of heap.
+        :target_bulk_bytes => {
+            :validate => :number,
+            :default => 9 * 1024 * 1024 # 9MiB
```

Comment:
Why not keep default as 20MiB for backward compatible and allow AWS OpenSearch Service users to set value?

Reply:
Because it doesn't work for many users giving problems that are hard to resolve. I think it's better if the plugin works out of the box.

Reply:
Changing default value will break backwards compatibility. Correct me if I am wrong, but if a user is sending bulk requests to a self managed cluster configured with `http.max_content_length` > 9 MiB, upgrading to a version with this change would cause the plugin to fail to send all data.

Instead of changing default, maybe a better solution would be to update [this documentation](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/managedomains-logstash.html).

Reply:
> Changing default value will break backwards compatibility. Correct me if I am wrong, but if a user is sending bulk requests to a self managed cluster configured with `http.max_content_length` > 9 MiB, upgrading to a version with this change would cause the plugin to fail to send all data.

Not at all. The only drawback is a slightly lower maximum throughput. But with this fix a user can tweak to their hearts content.

Reply:
One problem i could see is that we might be flushing bulk request frequent than before for size > 9 and < 20. Let's say, my total request size is 18 MB, with this change it might send two request than one, and proposal is to configure the value to say 20MB @msvticket can you correct if my understanding is wrong?

Reply:
It's correct. My opinion is that it is more important to prevent errors than preserving probably not that common cases.

Reply:
I think the only thing you're arguing about is whether to increment a minor version for this change, or a major version.

Sounds like the default was poorly chosen. If it causes bugs, then it's a bug fix, and changing the value of the default in a minor version is OK. Otherwise do the change in a major version increment. My take for this case is that it's ok to do it in a minor version increment.

Since it's configurable, you are preserving backwards compatibility. Behavior may change, but it's not broken. You will surprise some people who didn't read the release notes.